src/hash.rs

```diff
@@ -25,5 +25,5 @@
 /// Simple Hashable trait with single hash function.
 pub trait Hashable {
     /// Create a hash from this object
-    fn hash(&self) -> Vec<u8>;
+    fn hash(&self) -> [u8;32];
```

Comment:
@sdbondi is there a way to make the length generic?

Reply:
Yep const generic 

```rust
pub trait Hashable<const N: usize> {
    fn hash(&self) -> [u8; N];
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=914f81aee3bf6b19d923184bc7983800